### PR TITLE
Complete half-baked deliverable

### PR DESCRIPTION
**Pull request by ryanjeffrey**

### CODE REVIEW

File: lib/controllers/cats.js [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
 
-module.exports = Router();
+module.exports = Router()
+  .get('/:id', (req, res) => {
+    let match;
+    for (const cat of cats) {
+      if (cat.id === req.params.id) {
+        match = cat;
+      }
+    }
```
Comment (viccaruso) [inline lib/controllers/cats.js]: Using a `for` loop here works, but don't forget you have all the built-in array methods at your disposal. For this section you could have used the `.find()` method.

```js
const match = cats.find(cat => cat.id === req.params.id);

File: lib/controllers/cats.js [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+    return res.json(match);
+  })
+  .get('/', (req, res) => {
+    const filteredData = [];
+    for (const cat of cats) {
+      filteredData.push({ id: cat.id, name: cat.name });
+    }
```
Comment (viccaruso) [inline lib/controllers/cats.js]: Same story here. For this section you could have used the `.map()` method.
```js
const filteredData = cats.map((cat) => ({ id: cat.id, name: cat.name }));

Comment (ryanjeffrey) [reply lib/controllers/cats.js]: Awesome thanks Vic!

File: lib/controllers/cats.js (hunk 1 above)
Comment (ryanjeffrey) [reply lib/controllers/cats.js]: Sweet, good call.

**Pull request merged**